modules/generator/processor/spanmetrics/spanmetrics.go

```diff
@@ -70,7 +70,7 @@ func New(cfg Config, reg registry.Registry, filteredSpansCounter, invalidUTF8Cou
 		labels = append(labels, dimStatusMessage)
 	}
 
-	c := reclaimable.New(strutil.SanitizeLabelName, 10000)
+	c := reclaimable.New(strutil.SanitizeFullLabelName, 10000)
```

Comment:
this is so close!

my last concern is this change here. `SanitizieFullLabelName` creates a string builder and reallocs the string every time. `SanitizeLabelName` uses a regex to replace all invalid chars with `_`. we can either change it back or i'd need to see a benchmark of these two methods to confirm there isn't a regression here.

if you do write a benchmark let's make sure to bench the case where both substitutions and no substitutions are made on both funcs.